internal/provider/waypoint/resource_waypoint_action_config.go

```diff
@@ -0,0 +1,696 @@
+	if actionCfg.Request.Custom != nil {
+		data.Request.Custom = &customRequest{}
+		if actionCfg.Request.Custom.Method != nil {
+			methodString, err := ConvertMethodToStringType(*actionCfg.Request.Custom.Method)
+			if err != nil {
+				resp.Diagnostics.AddError(
+					"Unexpected HTTP Method",
+					"Expected GET, POST, PUT, DELETE, or PATCH. Please report this issue to the provider developers.",
+				)
+			} else {
+				data.Request.Custom.Method = methodString
+			}
+		}
+		if actionCfg.Request.Custom.Headers != nil {
+			for _, header := range actionCfg.Request.Custom.Headers {
+				headerMap[header.Key] = header.Value
+			}
+			if len(headerMap) > 0 {
+				data.Request.Custom.Headers, diags = types.MapValueFrom(ctx, types.StringType, headerMap)
+				resp.Diagnostics.Append(diags...)
+				if resp.Diagnostics.HasError() {
+					return
+				}
+			} else {
+				data.Request.Custom.Headers = types.MapNull(types.StringType)
+			}
+		}
+		if actionCfg.Request.Custom.URL != "" {
+			data.Request.Custom.URL = types.StringValue(actionCfg.Request.Custom.URL)
+		}
+		if actionCfg.Request.Custom.Body != "" {
+			data.Request.Custom.Body = types.StringValue(actionCfg.Request.Custom.Body)
+		}
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *ActionConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan *ActionConfigResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// get the current state as well, so we know the current name of the
+	// action config for reference during the update
+	var data *ActionConfigResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+
+	client := r.client
+	ns, err := getNamespaceByLocation(ctx, client, loc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"error getting namespace by location",
+			err.Error(),
+		)
+		return
+	}
+
+	modelBody := &waypoint_models.HashicorpCloudWaypointWaypointServiceUpdateActionConfigBody{
+		ActionConfig: &waypoint_models.HashicorpCloudWaypointActionConfig{
+			Request: &waypoint_models.HashicorpCloudWaypointActionConfigRequest{},
+		},
+	}
+
+	// These are the updated values
+	if !plan.Name.IsUnknown() {
+		modelBody.ActionConfig.Name = plan.Name.ValueString()
+	}
+	if !plan.Description.IsUnknown() {
+		modelBody.ActionConfig.Description = plan.Description.ValueString()
+	}
+	if !plan.ActionURL.IsUnknown() {
+		modelBody.ActionConfig.ActionURL = plan.ActionURL.ValueString()
+	}
```

Comment:
We should always read these and send them for update, I think? I just tried this branch out and confirmed that if you create a config with a description, and then update the tf configuration to remove the description, Terraform reports that it will remove the value but after successfully applying, the value remains unchanged. Every `terraform plan` afterwards shows a perpetual diff, because we never send the updated `""` value to the API